app/components/current-page.ts

```diff
@@ -0,0 +1,99 @@
+import type Store from '@ember-data/store';
+import type Book from 'collection-demo/models/book';
+import { tracked } from '@glimmer/tracking';
+import type { Collection } from '@ember-data/types';
+import { resource, resourceFactory } from 'ember-resources';
+
+export class AsyncContent<T> {
+  @tracked content: T | undefined;
+  @tracked isLoading = true;
+  @tracked error: string | Error | unknown | null = null;
+
+  get isInitiallyLoading() {
+    return this.isLoading && this.content === undefined;
+  }
+
+  get hasError() {
+    return this.error !== null;
+  }
+}
+
+interface FirstPageOptions {
+  url: string;
+  sort?: string | null;
+  filter?: string | null;
+  genre?: string | null;
+  author?: string | null;
+  page?: number | null;
+  limit?: number | null;
+}
+
+// TODO: inline to ember-resources?
+//  https://github.com/NullVoxPopuli/ember-resources/issues/945
+const unwrap = <T>(x: T | (() => T)): T =>
```

Comment:
not really needed here since we haven't implemented this resource to by type-complete from the template